polaris-react/src/components/Tag/Tag.scss

```diff
@@ -105,6 +105,10 @@ $button-size: 20px;
     min-height: $height;
     border-radius: var(--p-border-radius-2);
     padding-inline: calc(var(--p-space-1) + var(--p-space-05));
+
+    &.removable {
+      padding-inline-end: 0;
```

Comment:
Sneaky fix here; The `se23` `padding-inline` above accidentally overwrote the `.removable` style. Due to specificity, we have to duplicate it here so it's applied correctly.